### ISSUE TITLE
Support JDK 11

### ISSUE THREAD

**Issue by schroederc** [google/kythe#3095]
[X] Remove reliance on `-Xbootclasspath/p`
[ ] Fix JCTree incompatibility: http://hg.openjdk.java.net/jdk/jdk11/rev/f854b76b6a0c

**Comment by shahms** [comment]
There are a few other issues here:
1) javax.annotation.Generated moves to javax.annotation.processing.Generated in JDK9; this impacts:
  a) kythe/java/com/google/devtools/kythe/extractors/java/ProcessAnnotation.java 
  c) extraction tests (javax.annotations.Generated class files are included in more CUs as a result)
  b) and potentially the protobuf annotations
2) tests with newer features due to langtools changes
  a) compile langtools 11 for the oldest supported JDK
  b) or only enable the tests when running on a new JRE using the ambient langtools
3) whether or not we want to continue to support running on an older JDK
 a) if so, how do we test this

My preference would be to drop support for JRE 8 when we can, but I don't know how badly this will impact extant users.

**Comment by jaysachs** [comment]
My $0.02: we should not worry about JRE8 support as soon as possible.

**Comment by creachadair** [comment]
> My $0.02: we should not worry about JRE8 support as soon as possible.

+1. Can we disable support in some reversible way, push a release, and see if anyone squawks?

@benjyw would dropping 8 support cause you any problems?

**Comment by benjyw** [comment]
I don't think so, I'm in the process of getting our Kythe stuff working on JRE 11 (will need a few days to get a new Pants release with the necessary changes), I can report back here once it works and I'll be happy to be rid of 8 support.

**Comment by johnedmonds** [comment]
Wondering how backwards-compatible we could expect the output format of newer versions of Kythe to be. If we lost support for java 8 I suspect we could just package an older version of Kythe with a newer version and pick which one to run for each project to be indexed. It should work as long as there's no breaking changes to the indexer output.

**Comment by creachadair** [comment]
> Wondering how backwards-compatible we could expect the output format of newer versions of Kythe to be.

I don't expect the basic data format to change. The way the indexer encodes certain language features may change over time, however. The schema should reflect the current state of the model, but we haven't made any commitments about stability as yet.

> If we lost support for java 8 I suspect we could just package an older version of Kythe with a newer version and pick which one to run for each project to be indexed. It should work as long as there's no breaking changes to the indexer output.

For Java specifically, I would not _expect_ the model for existing language features to change much. The essential schema queries is pretty stable—jump to definition and cross-references, call graphs, and type hierarchy information for example.

To some extent it depends what queries you care about. It's possible to use Kythe data to answer questions _not_ protected by the schema, and in principle those queries could drift over time. If you have specific concerns you might want to start an issue or a discussion thread.

**Comment by benjyw** [comment]
PS We're fine with dropping support for running on JRE 8.

**Comment by shahms** [comment]
We're still stuck using JDK8 for a little while, sadly.  PR #3889 and #3892 fixed the biggest issues around `Generated` annotations.

The biggest impediment from needing to keep JDK8 support is that it complicates support for extraction/indexing of compilations which use modules.
